### PR TITLE
Feature: add leave node API

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

add leave node API in metasrv

## Changelog

- New Feature

## Related Issues

Fixes [Add API to remove a node from metasrv cluster · Issue #4312 · datafuselabs/databend](https://github.com/datafuselabs/databend/issues/4312)

### CODE REVIEW

File: metasrv/src/meta_service/meta_leader.rs [hunk 1]
```diff
@@ -151,6 +156,40 @@ impl<'a> MetaLeader<'a> {
...
+        membership.remove(&node_id);
+
+        let ent = LogEntry {
+            txid: None,
+            cmd: Cmd::RemoveNode { node_id },
+        };
+
+        self.write(ent.clone()).await?;
+
+        self.change_membership(membership).await
```
Comment (drmingdrmer) [inline metasrv/src/meta_service/meta_leader.rs:190]: If the first entry that removes a node is committed and applied but the second membership entry is not, the the network will find a absent node, which will cause a network error.

File: metasrv/src/meta_service/meta_leader.rs [hunk 2]
```diff
@@ -184,10 +184,11 @@ impl<'a> MetaLeader<'a> {
...
         };
-
-        self.write(ent.clone()).await?;
-
-        self.change_membership(membership).await
+        self.change_membership(membership).await?;
+        match self.write(ent.clone()).await {
+            Ok(_) => Ok(()),
+            Err(err) => Err(err),
+        }
```
Comment (drmingdrmer) [inline metasrv/src/meta_service/meta_leader.rs:191]: Does `ent` need a `clone()`?


And a simple way is :
```
self.write(ent.clone()).await?;
Ok(())
```

**Review by drmingdrmer — APPROVED** [review]
Everything else look good to me :DDD

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**